ee/dtr/admin/manage-webhooks/use-the-api.md

```diff
@@ -0,0 +1,281 @@
+---
+title: Manage webhooks via the API
+description: Learn how to create, configure, and test webhooks for DTR using the API.
+keywords: dtr, webhooks, api, registry
+---
+
+## Prerequisite
+
+See [Event types for webhooks](/ee/dtr/admin/manage-webhooks/index.md/#event-types-for-webhooks) for a complete list of event types you can trigger notifications for via the API.
+
+## API Base URL
+
+Your DTR hostname serves as the base URL for your API requests.
+
+## Subscribe to events
+
+To subscribe to events, send a `POST` request to
+`/api/v0/webhooks` with the following JSON payload:
+
+### Example Usage
+
+```
+{
+  "type": "TAG_PUSH",
+  "key": "foo/bar",
+  "endpoint": "https://example.com"
+}
+```
+
+The keys in the payload are:
+
+- `type`: The event type to subcribe to.
+- `key`: The namespace or repo to subscribe to. For example, "foo/bar" to subscribe to
+pushes to the "bar" repository within the namespace "foo".
+- `endpoint`: The URL to send the JSON payload to.
+
+Normal users **must** supply a "key" to scope a particular webhook event to
+a repository or a namespace. DTR admins can choose to omit this,
+meaning a POST event notification of your specified type will be sent for all DTR repositories and namespaces.
+
+### Receive a payload
+
+Whenever your specified event type occurs, DTR will send a POST request to the given
+endpoint with a JSON-encoded payload. The payload will always have the
+following wrapper:
+
+```
+{
+  "type": "...",
+  "createdAt": "2012-04-23T18:25:43.511Z",
+  "contents": {...}
+}
+```
+
+- `type` refers to the event type received at the specified subscription endpoint.  
+- `contents` refers to the payload of the event itself. Each event is different, therefore the
+structure of the JSON object in `contents` will change depending on the event
+type. See [Content structure](#content-structure) for more details.
+
+### Test payload subscriptions
+
+Before subscribing to an event, you can view and test your endpoints using
+fake data. To send a test payload, send `POST` request to
+`/api/v0/webhooks/test` with the following payload:
+
+```
+{
+  "type": "...",
+  "endpoint": "https://www.example.com/"
+}
+```
+
+Change `type` to the event type that you want to receive. DTR will then send
+an example payload to your specified endpoint. The example
+payload sent is always the same.
+
+## Content structure
+
+Comments after (`//`) are for informational purposes only, and the example payloads have been clipped for brevity.
+
+### Repository event content structure
+
+**Tag push**
+
+```
+{
+  "namespace": "",    // (string) namespace for the repository
```

Comment:
We can improve the presentation of this content later on -  but this is a quick fix that I'm hoping to get published this week.